AirportTicketBookingExercise/Data/Repository/BookingRepository.cs

```
@@ -0,0 +1,124 @@
+﻿using ATB.Data.Models;
+using ATB.Logic.Enums;
+using ATB.Logic.Extensions;
+using CsvHelper;
+using System.Globalization;
+
+namespace ATB.Data.Repository
+{
+    public class BookingRepository : IBookingRepository
+    {
+        private readonly string _bookingPath;
+
+        public BookingRepository(string bookingPath)
+        {
+            _bookingPath = bookingPath;
+        }
+
+        public List<Booking> GetAllBookings()
+        {
+            List<Booking> records = new List<Booking>();
+            try
+            {
+                using (var reader = new StreamReader(_bookingPath))
+                using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+                {
+                    csv.Context.RegisterClassMap<BookingMap>();
+                    records = csv.GetRecords<Booking>().ToList();
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
```

we shouldn't be printing anything from repos and services (data and logic)  layers. As mentioned before, App/UI layer is the one calling the data/logic and prints to the screen, We should have data and logic layers abstracted and just doing what it should do, so we can use this repo from other areas where we might not have CMD to print things to screen.

when you work later with APIs, we don't have CMD, we return data with APIs. so we shouldn't be expecting console.writeline here.

also here we are swallowing the exception, the exception should bubble up and lets the caller to handle the exception , as it's a generic exception.

---

please, review all files here and apply this comment